conans/client/build/autotools_environment.py

```diff
@@ -234,7 +234,10 @@ def make(self, args="", make_program=None, target=None, vars=None):
         if not self._conanfile.should_build:
             return
         conan_v2_error("build_type setting should be defined.", not self._build_type)
-        make_program = os.getenv("CONAN_MAKE_PROGRAM") or make_program or "make"
+        conan_make_program = os.getenv("CONAN_MAKE_PROGRAM")
+        if conan_make_program and self._win_bash:
+            conan_make_program = tools.unix_path(conan_make_program, path_flavor=self.subsystem)
```

Comment:
The problem with this is that this is based on auto-detection, not the real subsystem:
```python
self.subsystem = OSInfo().detect_windows_subsystem() if self._win_bash else None
```
There are reports already, that this is buggy, and might take the wrong subsystem. So this PR might break existing users, I'd prefer not to take the risk, and the users will probably have the workaround of specifying the ``CONAN_MAKE_PROGRAM`` directly in the subsystem path style? As it is very likely that make itself is in such subsytem, that might be good enough?

As we are moving fast towards the new AutotoolsDeps, AutotoolsToolchain, also re-doing the ``win_bash`` code, and basing it on user configuration instead of auto-detection, I'd prefer to wait and submit this issue there (if relevant, the ``CONAN_MAKE_PROGRAM`` will not be an env-var anymore, but ``[conf]`` instead).

Reply:
so you think correcting the path directly in https://github.com/conan-io/conan-center-index/blob/master/recipes/make/all/conanfile.py#L55 is better right ?

Reply:
Well, changing it in the recipe it won't be very pretty either, but maybe it has some smaller chance to break users.

Yes, I agree that it might be broken, but at least it is the "same broken". If it was broken for your case in the past it will continue to be broken, if it happen to work, even by luck, it will keep working. We have had too many experiences lately, breaking users with apparently insignificant changes (we have passed the limit of Hyrums Law clearly), so my current preference is try to avoid this kind of change unless very necessary, and move new changes, experimentation, etc to the new domains in preparation for Conan 2.0 (``conan.tools`` new stuff)

Reply:
actually it would not be correct to transform the path in [here](https://github.com/conan-io/conan-center-index/blob/master/recipes/make/all/conanfile.py#L55), because there is no way we can know at that point if make will be invoked in a bash environment or not.